File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;
class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        Map<String, Integer> map = new HashMap<> ();
        String[] keyarr;
        int value=0;
        for(String key : keymap){
            keyarr = key.split("");
            for(int i=0; i<keyarr.length; i++){
                if(map.get(keyarr[i])==null || map.get(keyarr[i])>i+1)
                    map.put(keyarr[i], i+1);
            }
        }
        for(int i=0; i<answer.length; i++){
            keyarr = targets[i].split("");
            for(int j=0; j<keyarr.length; j++){
                if(map.get(keyarr[j]) == null){
                    answer[i] = -1;
                    break;
                } else {
                    answer[i] += map.get(keyarr[j]);
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    let key = {};
    
    for(str of keymap){
        for(var i=0; i<str.length; i++){
            let ch = str[i];
            if(key[ch] == null) key[ch] = i+1;
            else key[ch] = Math.min(key[ch], i+1);
        }
    }
    
    targets.forEach(target => {
        var sum = 0;
        for(var i=0; i<target.length; i++) {
            let ch = target[i];
            if(key[ch] == null){
                sum = -1;
                break;
            }
            sum += key[ch];
        }
        answer.push(sum);
    })
    return answer;
}